### ISSUE TITLE
[Improvement] BatchNorm2d running stats 不在 parameters() 中，checkpoint 需手動處理

### ISSUE THREAD

**Issue by s9213712** [s9213712/minicnn#34]
## 問題描述

`src/minicnn/nn/layers.py` 中 `BatchNorm2d` 的 `running_mean` 和 `running_var` 是普通 numpy array，不是 `Parameter`：

```python
class BatchNorm2d(Module):
    def __init__(self, num_features, eps=1e-5, momentum=0.1):
        self.weight = self.add_parameter('weight', ...)   # ✓ 在 parameters() 中
        self.bias   = self.add_parameter('bias', ...)     # ✓ 在 parameters() 中
        self.running_mean = np.zeros(...)                 # ✗ 不在 parameters() 中
        self.running_var  = np.ones(...)                  # ✗ 不在 parameters() 中
```

## 影響

- **嚴重性**: Major（影響模型可重現性）
- 使用 `checkpoint` 系統保存/載入模型時，若未特別處理 `running_mean`/`running_var`，推理階段的 BatchNorm 行為會與訓練結束時不同
- 從 checkpoint 恢復訓練時，running stats 從零開始，前幾個 batch 的 BN 行為異常

## 建議修復

在 checkpoint 保存/載入時包含 running stats：

```python
# 在 checkpoint 邏輯中
state = {
    'params': {name: p.data for name, p in model.named_parameters()},
    'buffers': {
        'bn_running_mean': layer.running_mean,
        'bn_running_var':  layer.running_var,
    }
}
```

或在 `Module` 基類加入 `register_buffer()` 機制（類似 PyTorch）。

## 相關位置

- `src/minicnn/nn/layers.py:108-128` — `BatchNorm2d.__init__()`
- `src/minicnn/training/checkpoints.py` — checkpoint 儲存邏輯需確認
